File: week-4/gray-code/gray-code.py
class Solution:
    def grayCode(self, n: int) -> List[int]:
        res = [0]
        for i in range(0, n):
            size = len(res)
            for j in range(size-1, -1, -1):
                res.append(res[j] | 1<<i)
            
        return res

File: week-4/gray-code/gray-code.js
/**
 * @param {number} n
 * @return {number[]}
 */
var grayCode = function(n) {
    const res = [0];
    for (let i=0; i<n; i++) {
        const size = res.length;
        for (let j=size-1; j>=0; j--)
            res.push(res[j] | 1<<i);
    }
    
    return res;
};
